internal/generator/renderValues.go

```diff
@@ -0,0 +1,41 @@
+package generator
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"gopkg.in/yaml.v3"
+
+	"github.com/nestoca/joy/api/v1alpha1"
+)
+
+// RenderValues renders the values of the given release and produces a yaml string, processing any go template
+// directives found in the values.
+func RenderValues(release *v1alpha1.Release) (string, error) {
+	valuesStr, err := yaml.Marshal(release.Spec.Values)
+	if err != nil {
+		return "", fmt.Errorf("marshalling release values: %w", err)
+	}
+
+	tpl, err := template.New("values").Parse(string(valuesStr))
+	if err != nil {
+		return "", fmt.Errorf("parsing values template: %w", err)
+	}
+
+	tpl.Option("missingkey=error")
+
+	var result bytes.Buffer
+	type TemplateData struct {
+		Release *v1alpha1.Release
```

Comment:
We should probably already also inject the `Environment` here, even if we don't currently use it. It can be super useful to reference it from the release values (for example, to dynamically include the env name in an ingress url, which I did in the past).  It also allows to define custom variables in the `Environment.spec.values` that can be reused across all releases, without duplication.

Reply:
Sure I can add it, I had hesitated initially since technically the env data is reachable via `Release.Environment...` but I can add it to the top level if you feel thats better.